src/Generator.ts

```diff
@@ -0,0 +1,196 @@
+import type {
+  EntryType,
+  DirectoryContent,
+  HeaderOptions,
+  ReadFileOptions,
+  WalkDirectoryOptions,
+  TarOptions,
+} from './types';
+import fs from 'fs';
+import path from 'path';
+import { EntryTypes } from './types';
+import * as errors from './errors';
```

Comment:
I do not believe it is necessary to import `fs`. Note that we want this to be VIRTUAL. That means nothing from `fs` should be used. We want this library to work without dependency on nodejs OS primitives.